tests/jballerina-unit-test/src/test/resources/test-src/expressions/errorconstructorexpr/error-constructor-expr-negative.bal

```diff
@@ -62,3 +62,6 @@ type ErrorU3 map<int>|map<Blah>|map<string>;
 error uc1 = error ErrorU1("Hello");
 error uc2 = error ErrorU2("Hello");
 error uc3 = error ErrorU3("Hello");
+
+int|string uc4 = error ErrorA("msg");
```

Comment:
For this particular case, I think the previous error may be more accurate. When we say `error E(...)` we are saying the type of the constructed error is `E` right? So I think we should continue to say `expected 'T' found 'E'` here. WDYT?

Reply:
Agree. Check the latest commit.
70b674a78d76dede5031e3fa2882de65f808a48d

Reply:
But does the `found 'error'` part make sense when we say `a type compatible with error constructor expressions not found`?

Can't we do
1. if the user provides a `error-type-reference`, use the "incompatible types: expected 'T', found 'E'" error
2. else, use "a type compatible with error constructor expressions not found in type 'T'"